android/automation.md

```diff
@@ -0,0 +1,175 @@
+# Automation and continuous integration systems
+
+Our Android projects share a similar automation model:
+- Taskcluster is our primary CI: we use it for as much automation as
+possible, including our core builds and releases, because it's a
+flexible first party service we can rely on.
+- Travis is secondary: we use it to get automation working quickly or get
+support for tools that won't work on Taskcluster.
+
+## Taskcluster
+
+[Taskcluster][tc] is a generic task execution framework developed and used by Mozilla to automate continuous integration and release processes.
+
+We currently use Taskcluster to:
+* Build, test and run code tools for all master commits and pull requests
+(including contributor PRs). See [Code quality tools](#code-quality-tools)
+for additional information.
+* Build (unsigned) release builds whenever a release is created on GitHub
+* Automatically import new translations from the L10N repository.
+
+To see a Taskcluster configuration for your project, see
+`<project-root>/.taskcluster.yml` (here's an
+[example in focus-android][tc yml]).
+
+Taskcluster runs using docker: see our [docker guide](docker_guide.md) for
+more information.
+
+## Travis
+
+[Travis] is an easy-to-configure continuous integration environment with
+great support for GitHub.
+
+We currently use Travis to:
+* Build, test and run code tools for all master commits and pull requests (including contributor PRs).
+* Execute sputnik for posting automated code feedback in pull requests
+* Execute codecov for monitoring our code coverage
+
+To see a travis configuration for your project, see `<project-root>/.travis.yml` (here's an
+[example in focus-android][travis yml]).
+
+You can also find build logs on the travis website, e.g. for focus-android: https://travis-ci.org/mozilla-mobile/focus-android
+
+# Code quality tools
+
+For Java/Kotlin projects, we like to run the following tools:
+* findbugs
+* PMD
+* Android lint
+* checkstyle
+* [detekt](https://github.com/arturbosch/detekt)
+* [ktlint](https://github.com/shyiko/ktlint)
+
+To see what tools are currently run in automation for your project, see
+the project's `<project-root>/.taskcluster.yml`
+(here's an [example in focus-android][tc yml tools]).
+
+# Third-party services
+
+## Sputnik
+
+Sputnik (riggered from travis) runs findbugs, PMD and checkstyle on code attached to pull requests and posts in-line comments whenever violations are found.
+
+* https://sputnik.ci
+
+## Codecov
+
+Codecov evaluates the code coverage reports generated by our build and tracks the coverage over time. It warns when merging pull requests would lead to dropping the coverage below a specified threshold.
+
+* https://codecov.io/gh
+* [focus-android code coverage over time](https://codecov.io/gh/mozilla-mobile/focus-android/branch/master)
+
+# Automation tasks
+
+## Pull request pipeline
+
+![](https://i.imgur.com/SJ9zSs8.png)
+
+## Release pipeline
```

Comment:
This section is heavily outdated and very Focus-focused (lol). I guess instead we want to shape this into something like: If you start a new project: That's how we do it.

Reply:
tbh, I was kind of hoping to land this partially incorrect and that someone would update it for me (because I don't actually know how it all works). :) But I'll do my best to adapt to your suggestions.